src/main/java/com/microsoft/azure/vmagent/AzureVMManagementServiceDelegate.java

```diff
@@ -1132,6 +1173,18 @@ public Void call() throws Exception {
                 }
             }, new NoRetryStrategy());
         }
+
+    }
+
+    public static void removeImage(final Azure azureClient, final String vmName, final String resourceGroupName) {
+        List<VirtualMachineCustomImage> customImages = azureClient.virtualMachineCustomImages().listByGroup(resourceGroupName);
+        for (VirtualMachineCustomImage image : customImages) {
+            String prefix = StringUtils.substringBefore(image.name(), "Image");
+            if (StringUtils.contains(vmName, prefix)) {
```

Comment:
Because actually we can't get the it's name here directly. `vmName` here is `vmNameBase+copyIndex`, but image haven't an attribute named` copyIndex`. So, vmName maybe `Jenkins43a021` but image name is `Jenkins43a02Image `. Note that image name haven't the last `1`, and we don't know vmName's copyIndex here directly.